mdata/aggregator.go

```diff
@@ -100,19 +100,19 @@ func (agg *Aggregator) flush() {
 func (agg *Aggregator) Add(ts uint32, val float64) {
 	boundary := AggBoundary(ts, agg.span)
 
-	if boundary == agg.currentBoundary {
-		agg.agg.Add(val)
-		if ts == boundary {
-			agg.flush()
-		}
+	if boundary < agg.currentBoundary {
+		return
```

Comment:
Shouldnt this never happen?

Reply:
that was true until a few days ago.
due to https://github.com/grafana/metrictank/pull/1382/commits/16782efed94e07a8bbb3007d4114ee0fcae94ef3 it is no longer true: to honor ingest-from for aggregators, AggMetric passes the point on to the aggregator at a point that is quite a bit before the part where it can make sure the data is in order.